pkg/platform/kube/platform.go

```diff
@@ -1948,6 +1952,88 @@ func (p *Platform) validateServiceType(functionConfig *functionconfig.Config) er
 	}
 }
 
+func (p *Platform) validateServiceAccount(ctx context.Context, functionConfig *functionconfig.Config) error {
+	// if function does not have service account specified, skip validation
+	// it will be enriched later by the controller
+	if functionConfig.Spec.ServiceAccount == "" {
+		return nil
+	}
+
+	// check if the service account is allowed
+	return p.isServiceAccountAllowed(ctx, functionConfig)
+}
+
+func (p *Platform) isServiceAccountAllowed(ctx context.Context, functionConfig *functionconfig.Config) error {
+	projectName, ok := functionConfig.Meta.Labels[common.NuclioResourceLabelKeyProjectName]
+	if !ok {
+		return errors.New("Function does not have a project label, cannot validate service account")
+	}
+
+	// fetch the secret from Kubernetes
+	secret, err := p.getProjectSecret(ctx, projectName, functionConfig.Meta.Namespace)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get project secret. Project: %s", projectName)
+	}
+
+	if secret == nil {
+		// if the secret does not exist, skip validation
+		return nil
+	}
+
+	// get the key name to use for allowed service accounts
+	if p.Config.Kube.ProjectSecretAllowedServiceAccountsKey == "" {
+		p.Logger.DebugWithCtx(ctx, "Skipping service account secret validation as no `projectSecretAllowedServiceAccountsKey` is configured")
+		return nil
+	}
+
+	allowedRaw, ok := secret.Data[p.Config.Kube.ProjectSecretAllowedServiceAccountsKey]
+	if !ok {
+		// no restriction , so allow any SA
+		return nil
+	}
+
+	allowedSAs := strings.Split(string(allowedRaw), ",")
+
+	// trim spaces and check membership
+	requestedSA := strings.TrimSpace(functionConfig.Spec.ServiceAccount)
+	for _, sa := range allowedSAs {
+		if strings.TrimSpace(sa) == requestedSA {
+			return nil // valid SA
+		}
+	}
```

Comment:
is this case-sensitive?
IIRC kubernetes resource names are not, but the values in the secret / function config can be - we can `lower` all of them.